rxnetty/src/main/java/io/reactivex/netty/client/ConnectionPoolImpl.java

```diff
@@ -110,54 +120,73 @@ public ConnectionPoolImpl(RxClient.ServerInfo serverInfo, PoolConfig poolConfig,
     @Override
     public Observable<ObservableConnection<I, O>> acquire() {
 
-        if (isShutdown.get()) {
+        if (isShutdownRequested.get()) {
```

Comment:
Yes I think doing it on subscription is enough (and correct) as subscriptions are lazy. So, generally state checks are better on subscription vis-a-vis on creation.